piksi_tools/console/update_view.py

```diff
@@ -497,6 +538,58 @@ def _download_firmware(self):
                 )
             self.downloading = False
             return
+        """ Download latest firmware from swiftnav.com. """
+        self._write('')
+
+        # Check that we received the index file from the website.
+        if self.update_dl is None or self.update_dl.index is None:
+            self._write("Error: Can't download firmware files")
+            return
+
+        self.downloading = True
+        status = 'Downloading Latest Firmware...'
+        self.stm_fw.clear(status)
+        self._write(status)
+
+    def _download_console(self, path=None):
+        """ Download latest firmware from swiftnav.com. """
+        self._write('')
+
+        # Check that we received the index file from the website.
+        if self.update_dl is None or self.update_dl.index is None:
+            self._write("Error: Can't download firmware files")
+            return
+
+        status = 'Downloading Latest Console to {}'.format(path)
+        self._write(status)
+
+        # Get console files from Swift Nav's website for correct platform
+        if 'console' in self.update_dl.index[self.piksi_hw_rev]:
+            try:
```

Comment:
The KeyError and AttributeError blocks encompass very broad errors, the specific situations where these exceptions render into specific error cases should have try/catch blocks around them, and we should throw a subclass of RuntimeError with an appropriate message.